app/privkeylock/privkeylock.go

```diff
@@ -3,65 +3,172 @@
 package privkeylock
 
 import (
+	"context"
+	"encoding/json"
 	"os"
+	"time"
 
 	"github.com/obolnetwork/charon/app/errors"
 	"github.com/obolnetwork/charon/app/z"
 )
 
-// New creates a private key lock file in path, writing contextStr in it.
-// If a private key lock file exists at path New returns an error, a clean-up function otherwise.
-func New(path, contextStr string) (func() error, error) {
+var lockfileGracePeriod = func() time.Duration {
+	return 5 * time.Second
+}
+
+type privkeyLockCtx struct {
```

Comment:
not sure about naming:
 - it stutters with package name
 - Ctx is usually used for context.Context
 
 what about `fileContent` or `metadata`?

Reply:
`metadata` sounds good!.